src/agentscope/rag/search_knowledge.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+"""
+This module is an integration of the Llama index RAG
+into AgentScope package
+"""
+import json
+import os.path
+from typing import Any, Optional, List
+
+from llama_index.core.data_structs import Node
+
+# search
+from llama_index.core.schema import NodeWithScore
+from loguru import logger
+
+from agentscope.constants import (
+    DEFAULT_CHUNK_SIZE,
+)
+from agentscope.service import load_web
+from agentscope.utils.token_utils import count_openai_token
+from agentscope.service import bing_search
+from agentscope.rag.knowledge import Knowledge
+
+
+class BingKnowledge(Knowledge):
+    """
+    This class is a wrapper with the Bing search api .
+    """
+
+    knowledge_type = "bing_knowledge"
+
+    def __init__(
+        self,
+        knowledge_id: str,
+        knowledge_config: Optional[dict] = None,
+        to_load_web: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        """
+        This class is a knowledge wrapper using Bing Search to
+        retrieve information online.
+
+        Args:
+            knowledge_id (str):
+                The id of the RAG knowledge unit.
+            knowledge_config (dict):
+                The configuration for using online search
+            load_web (bool):
+                whether to load the web page content. Currently only supports
+                loading content in "p" tags
+        """
+        super().__init__(
+            knowledge_id=knowledge_id,
+            knowledge_config=knowledge_config,
+            **kwargs,
+        )
+        self.to_load_web = to_load_web
+
+    def _init_rag(
+        self,
+        **kwargs: Any,
+    ) -> Any:
+        pass
+
+    def retrieve(
+        self,
+        query: Any,
+        similarity_top_k: int = None,
+        to_list_strs: bool = False,
+        **kwargs: Any,
+    ) -> List[Any]:
+        """
+        retrieve content from bing search and format them in list
+        """
+        search_config = self.knowledge_config.get("bing_search_config", {})
```

Comment:
What's bing search config and how to create a bing search config? Cannot find any details in this PR.